code/datums/status_effects/debuffs.dm

```diff
@@ -958,3 +957,24 @@
 /datum/status_effect/cloudstruck/Destroy()
 	. = ..()
 	QDEL_NULL(mob_overlay)
+
+/// Status effect knocks the owner to the floor, immobilising and incapacitating them while they are prone.
+/datum/status_effect/incapacitating/wormed
+	id = "wormed"
+
+/datum/status_effect/incapacitating/wormed/on_apply()
+	. = ..()
+	if(!.)
+		return
+	ADD_TRAIT(owner, TRAIT_FLOORED, TRAIT_STATUS_EFFECT(id))
+	ADD_TRAIT(owner, TRAIT_INCAPACITATED, TRAIT_STATUS_EFFECT(id))
+	ADD_TRAIT(owner, TRAIT_IMMOBILIZED, TRAIT_STATUS_EFFECT(id))
+	ADD_TRAIT(owner, TRAIT_HANDS_BLOCKED, TRAIT_STATUS_EFFECT(id))
```

Comment:
Wait, I thought you wanted to drop them to the floor and impede their hand usage. I didn't notice you wanted to also immobilize them, else you just have the paralyzed condition, don't you?